File: app/src/main/java/com/soft/credit911/ui/Chat/ChatContract.java
package com.soft.credit911.ui.Chat;

import com.soft.credit911.datamodel.ChatObject;

import java.util.ArrayList;

public interface ChatContract {
    interface View {
        void notifyAdapterObjectAdded(int position);
        void scrollChatDown();
    }

    interface Presenter {

        void attachView(ChatContract.View view);

        ArrayList<ChatObject> getChatObjects();

        void onEditTextActionDone(String inputText);
    }
}

File: app/src/main/java/com/soft/credit911/ui/Chat/ChatContract.kt
package com.soft.credit911.ui.Chat

import com.soft.credit911.datamodel.ChatObject
import java.util.*

interface ChatContract {
    interface View {
        fun notifyAdapterObjectAdded(position: Int)
        fun scrollChatDown()
    }

    interface Presenter {
        fun attachView(view: View?)
        //val chatObjects: ArrayList<ChatObject>?
        fun onEditTextActionDone(inputText: String?)
       // fun getChatObjects(): ArrayList<ChatObject>?
    }
}